src/lib/m2/index.js

```diff
@@ -116,5 +128,29 @@ export default new r.Struct({
   }),
   unknown2: new r.Optional(r.uint32le, function() {
     return this.flags === 8;
-  })
+  }),
+
+  isAnimated: function() {
+    let animated = false;
+
+    this.bones.forEach((bone) => {
+      if (bone.isAnimated) {
+        animated = true;
+      }
+    });
+
+    this.transparencies.forEach((transparency) => {
+      if (transparency.isAnimated) {
+        animated = true;
+      }
+    });
+
+    this.colors.forEach((color) => {
+      if (color.color.isAnimated || color.alpha.isAnimated) {
+        animated = true;
+      }
+    });
+
+    return animated;
+  }
```

Comment:
Is there a way to return from a parent function from within the function passed to forEach? I'd love to early return on, say, lines 144 and 150 directly.

In general, the speed benefit will only apply to when restructure is parsing, since it actually converts top-level functions to values from what I can tell.